components/raftstore/src/store/msg.rs

```diff
@@ -836,6 +837,10 @@ impl<EK: KvEngine> fmt::Debug for PeerMsg<EK> {
 }
 
 impl<EK: KvEngine> PeerMsg<EK> {
+    pub fn discriminant(&self) -> u8 {
+        unsafe { *(self as *const Self as *const u8) }
+    }
```

Comment:
Could you add some comments about how it works?

Reply:
PTAL again